thread.c

```diff
@@ -1108,129 +1122,152 @@ rb_thread_create_ractor(rb_ractor_t *g, VALUE args, VALUE proc)
 
 
 struct join_arg {
-    rb_thread_t *target, *waiting;
-    rb_hrtime_t *limit;
+    struct rb_waiting_list *waiting_list;
```

Comment:
Why is this a `rb_waiting_list*` and not just a single thread?

Reply:
It's a linked list node.

Reply:
That's a confusing name, `rb_waiting_list *` sounds like a list of a list.
I think passing `rb_thread_t *waiting/waiter` is enough here, no?

Reply:
OK, so it's just a thread+fiber + wait-queue/list-node.
I think we should use a better name for that. `rb_thread_list_t` before was not great but at least here it was passed as a single `rb_thread_t` which was clear.

Maybe `rb_waiting_thread_fiber*` or `rb_waiting_entry*`?
All such things form a list, each of them is not a list conceptually (but a node/entry of the list).

Reply:
I'm fine with this terminology and it's used somewhat consistently throughout the code base. With a linked list, every node is also a list. This is especially true for a singly linked list.